src/m365/search/commands/externalconnection/externalconnection-add.ts

```diff
@@ -0,0 +1,144 @@
+import { Logger } from "../../../../cli";
+import { CommandOption } from "../../../../Command";
+import GlobalOptions from "../../../../GlobalOptions";
+import request from "../../../../request";
+import GraphCommand from "../../../base/GraphCommand";
+import commands from "../../commands";
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  name: string;
+  description?: string;
+  authorizedAppIds?: string;
+}
+
+class SearchExternalConnectionAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.EXTERNALCONNECTION_ADD;
+  }
+
+  public get description(): string {
+    return "Adds a new External Connection for Microsoft Search";
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ["id", "name", "description"];
+  }
+
+  public commandAction(
+    logger: Logger,
+    args: CommandArgs,
+    cb: () => void
+  ): void {
+
+    let appIds: string[] = [];
+
+    if (
+      args.options.authorizedAppIds !== undefined &&
+      args.options.authorizedAppIds !== ""
+    ) {
+      appIds = args.options.authorizedAppIds?.split(",");
+    }
+
+    const commandData = {
+      id: args.options.id,
+      name: args.options.name,
+      description: args.options.description,
+      configuration: {
+        authorizedAppIds: appIds
+      }
+    };
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/external/connections`,
+      headers: {
+        accept: "application/json;odata.metadata=none"
+      },
+      responseType: "json",
+      data: commandData
+    };
+
+    request.post(requestOptions).then(
+      (_) => cb(),
+      (err: any) => {
+        this.handleRejectedODataJsonPromise(err, logger, cb);
+      }
+    );
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: "--id <id>"
+      },
+      {
+        option: "--name <name>"
+      },
+      {
+        option: "--description <description>"
```

Comment:
`description` property is defined as optional in the `Options` interface but here is defined as mandatory.

```suggestion
        option: "--description [description]"
```